src/Core/3DTiles/C3DTileset.js

```diff
@@ -0,0 +1,112 @@
+ *  [Tileset](https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/specification/schema/tileset.schema.json).
+ *  @property {boolean} isC3DTileset - Used by 3D Tiles extensions
+ *  (e.g. {@link C3DTBatchTableHierarchyExtension}) to know in which context
+ *  (i.e. for which 3D Tiles class) the parsing of the extension should be done.
+ *  This value is set to true by default.
+ *  @property {object} asset - Generic information about the tileset, see
+ *  [asset specification]https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/asset.schema.json
+ *  @property {object} properties - Properties associated with the tileset, see
+ *  [tileset specification](https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/tileset.schema.json#L11)
+ *  @property {number} geometricError - see [tileset
+ *  specification](https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/tileset.schema.json#L18)
+ *  @propertu {array} extensionsUsed - see [tileset specification](https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/tileset.schema.json#L27)
+ *  @property {array} extensionsRequired - see [tileset specification](https://github.com/CesiumGS/3d-tiles/blob/master/specification/schema/tileset.schema.json#L36)
+ *  @propertiy {object[]} tiles - an array holding all the
+ *  [tiles](https://github.com/AnalyticalGraphicsInc/3d-tiles/blob/master/specification/schema/tile.schema.json)
+ *  (not their content which is stored in the attribute object3d of the
+ *  layer). This list represents a flattened tileset.
+ *  @property {object} extensions - Extensions of the tileset in the form:
+ * {extensioName1: extensionObject1, extensioName2: extensionObject2, ...}
+ */
+class C3DTileset {
+    constructor(json, baseURL, registeredExtensions) {
+        this.isC3DTileset = true;
+
+        this.asset = json.asset;
+        this.properties = json.properties;
+        this.geometricError = json.geometricError;
+        this.extensionsUsed = json.extensionsUsed;
+        this.extensionsRequired = json.extensionsRequired;
+        this.tiles = [];
+        this.parseTiles(json.root, baseURL, undefined, registeredExtensions);
+
+        if (json.extensions) {
+            this.extensions =
+                registeredExtensions.parseExtensions(json.extensions, this);
+        }
+    }
+
+    /**
+     * Recursion on the 3DTiles tileset (which is a tree) to parse the tiles
+     * (nodes of the tree).
+     * @param {object} tile - current tile
+     * @param {string} baseURL - url of the source tileset and tiles
+     * @param {object} parent - parent tile (used for computing the transform
+     * matrix from local to global coordinates)
+     * @param {object} registeredExtensions - 3D Tiles extensions registered
+     * in the C3DTilesLayer (see {@link C3DTExtensions}
+     */
+    parseTiles(tile, baseURL, parent, registeredExtensions) {
+        const inverseTileTransform = new THREE.Matrix4();
```

Comment:
You mean before the class declaration ? What would be the benefit ?

Reply:
yes, before the class declaration because `inverseTileTransform` is instanced each call function and increases the memory so decreases performance to clean memory with the Garbage Collector.

Reply:
Ok, I'll move it. Just for my personal curiosity, do you know if this gain is quantifiable or if it has been quantified on simple cases?

Reply:
With tools performances in Chrome you can track the garbage collector time (Major GC).
Instance Object only used in scope function increase major GC time.

Reply:
Ok, thanks.